src/bin/sol-fbp-generator/main.c

```diff
@@ -952,12 +952,57 @@ handle_json_path(const char *path)
         dup_path = sol_arena_strdup(str_arena, path);
         SOL_NULL_CHECK(dup_path, false);
 
-        sol_ptr_vector_append(&args.json_files, dup_path);
+        if (sol_ptr_vector_append(&args.json_files, dup_path) < 0) {
+            SOL_ERR("Couldn't handle json path '%s': %s\n", path, sol_util_strerrora(errno));
+            return false;
+        }
+    }
+
+    return true;
+}
+
+static bool
+handle_include_path(char *path)
+{
+    struct stat s;
+    char *dup_path;
+
+    if (stat(path, &s) != 0)
+        return false;
+
+    if (S_ISDIR(s.st_mode))
+        dup_path = sol_arena_strdup(str_arena, path);
+    else if (S_ISREG(s.st_mode))
+        dup_path = sol_arena_strdup(str_arena, dirname(path));
+
+    SOL_NULL_CHECK(dup_path, false);
+    if (sol_ptr_vector_append(&args.include_paths, dup_path) < 0) {
+        SOL_ERR("Couldn't handle include path '%s': %s\n", path, sol_util_strerrora(errno));
+        return false;
     }
 
     return true;
 }
 
+static bool
+handle_include_dir(char *fullpath, const char *basename)
+{
+    struct stat s;
+    const char *p;
+    uint16_t i;
+    int err;
+
+    SOL_PTR_VECTOR_FOREACH_IDX (&args.include_paths, p, i) {
+        err = snprintf(fullpath, PATH_MAX, "%s/%s", p, basename);
+        SOL_INT_CHECK(err, < 0, false);
```

Comment:
You should check if the returned value is > fullpath as well.
e.g. `if (len < 0 || len > (int)sizeof(fullpath))`